File: bank-app/app/src/main/java/com/android/bankapp/statements/presenter/StatementsPresenter.java
package com.android.bankapp.statements.presenter;

import android.util.Log;

import com.android.bankapp.service.BankService;
import com.android.bankapp.service.ServiceGenerator;
import com.android.bankapp.statements.model.StatementResponse;
import com.android.bankapp.statements.view.StatementsActivity;
import com.android.bankapp.util.UserStateUtil;

import java.lang.ref.WeakReference;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class StatementsPresenter implements StatementPresenterInput {

    private static final String TAG = "StatementsPresenter";
    public WeakReference<StatementsActivity> output;
    private BankService service;

    public StatementsPresenter() {
        service = ServiceGenerator.generateService(BankService.class);
    }

    @Override
    public void loadData() {

        int id = UserStateUtil.getUserAccount().getUserId();

        Call<StatementResponse> call = service.loadStatement(id);

        call.enqueue(new Callback<StatementResponse>() {
            @Override
            public void onResponse(Call<StatementResponse> call, Response<StatementResponse> response) {
                if (response.body() != null) {
                    if (response.body().getStatementList() != null) {
                        output.get().dataLoaded(response.body().getStatementList());
                    } else {
                        output.get().errorLoadData();
                    }
                } else {
                    output.get().errorLoadData();
                }
            }

            @Override
            public void onFailure(Call<StatementResponse> call, Throwable t) {
                output.get().errorLoadData();
                Log.e(TAG, t.getMessage());
            }
        });

    }
}

File: app/src/main/java/com/example/testesantander/ui/statements/StatementsPresenter.kt
package com.example.testesantander.ui.statements

import com.example.testesantander.domain.model.StatementsData
import com.example.testesantander.domain.model.StatementsResponse
import com.example.testesantander.domain.usecase.IGetStatementsUseCase
import com.example.testesantander.mvp.BasePresenter
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class StatementsPresenter(private val getStatementsUseCase: IGetStatementsUseCase): BasePresenter<StatementsContract.View>(), StatementsContract.Presenter{

    private var statements: StatementsResponse? = null
    private lateinit var mStatementList: Array<StatementsData>

    override fun getStatementsList(){
        getStatementsUseCase.execute().enqueue(object : Callback<StatementsResponse>{
            override fun onFailure(call: Call<StatementsResponse>, t: Throwable) {

            }

            override fun onResponse(call: Call<StatementsResponse>, response: Response<StatementsResponse>) {
                statements = response.body()
                mStatementList = statements!!.statementList
                mView?.getList(mStatementList)
            }

        })
    }

}